### PR TITLE
Rewires library page. Adds loader.

### PR DESCRIPTION
**Pull request by ZachBaird**
Fixes MOD-384

- Adds AnimatedLogo to Library page as it loads.
- Adds play and stop functionality to instances.
- Splits library into instances and modpacks based on if linked_project_id is present.

Notes:

- Much of this code is similar to the wire-up-homepage code. Could cause merge conflict later.
- The old `kill_by_pid` process helper has changed to `kill_by_uuid`.
- Trying to play results in a stack overflow from tokio.

### CODE REVIEW

File: theseus_gui/src/pages/Library.vue [hunk 1]
```diff
@@ -1,15 +1,36 @@
...
   </div>
 </template>
+
+<style lang="scss" scoped>
+.loader {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 100%;
+}
```
Comment (Geometrically) [inline theseus_gui/src/pages/Library.vue:35]: no need for loading stuff like this here, see my other reviews

File: theseus_gui/src/pages/Library.vue [hunk 2]
```diff
@@ -1,15 +1,36 @@
...
+let modpacks
+
+onMounted(async () => {
+  loading.value = true
+  const profiles = await list()
+  instances = shallowRef(Object.values(profiles).filter((prof) => !prof.metadata.linked_project_id))
+  modpacks = shallowRef(Object.values(profiles).filter((prof) => prof.metadata.linked_project_id))
+
+  loading.value = false
+})
```
Comment (Geometrically) [inline theseus_gui/src/pages/Library.vue:18]: should use script setup

File: theseus_gui/src/components/ui/Instance.vue [hunk 3]
```diff
@@ -43,18 +79,27 @@ const props = defineProps({
...
-    .cta {
+    .install {
       opacity: 1;
       bottom: 4.5rem;
     }
   }
 
+  .install {
+    background: var(--color-brand);
+    z-index: 55;
```
Comment (Geometrically) [inline theseus_gui/src/components/ui/Instance.vue:90]: random z indexes shouldnt be used

File: theseus_gui/src/components/ui/Instance.vue [hunk 4]
```diff
@@ -12,22 +15,55 @@ const props = defineProps({
...
+
+const stop = async (e) => {
+  console.log('stopping')
+  e.stopPropagation()
+  await kill_by_uuid(uuid.value)
+  stopBtn.value.style.opacity = 0
+  stopBtn.value.style.bottom = '0'
+  stopBtn.value.style.display = 'none'
+  playBtn.value.style.display = 'flex'
+}
```
Comment (Geometrically) [inline theseus_gui/src/components/ui/Instance.vue:51]: see my other comment about this

File: theseus_gui/src/components/ui/Instance.vue [hunk 5]
```diff
@@ -12,22 +15,55 @@ const props = defineProps({
...
+const playBtn = ref(null)
+
+const uuid = ref(null)
+
+const router = useRouter()
+
+const seeInstance = async () => {
+  const instancePath = `/instance/${encodeURIComponent(props.instance.path)}`
+  await router.push(instancePath)
+}
```
Comment (Geometrically) [inline theseus_gui/src/components/ui/Instance.vue:29]: this can probably be inlined

**Review by Geometrically — CHANGES_REQUESTED** [review]

File: theseus_gui/src/components/ui/Instance.vue (hunk 5 above)
Comment (ZachBaird) [reply theseus_gui/src/components/ui/Instance.vue:29]: The e.stopPropagation during the Play click didn't always stop the RouterLink (that existed on the card originally) from navigating. This provided a much more consistent experience.

Comment (Geometrically) [reply theseus_gui/src/components/ui/Instance.vue]: By inlined I mean it can just be a lambda rather than a seperate function

**Review by Geometrically — CHANGES_REQUESTED** [review]
This PR looks good but I think it should just be rebased to the home page PR since most of the diff is shared with it

**Pull request closed without merging**